### ISSUE TITLE
FluxBB and PHP 8

### ISSUE THREAD

**Issue by philwareham** [textpattern/textpattern-forum#434]
As discussed privately already, we need a route forward for our forum so it runs on our latest server instance (i.e. PHP 8). The current stable release of FluxBB is only compatible with PHP up to v7.3. I don’t particularly want to move to a totally new platform as the software is great for our needs - and we have well over a decade of posts on it. Plus I’ve themed it to fit perfectly into our site aesthetic and performance is super-quick.

The two options I can see are:

1. Wait on a possible new release of official FluxBB (which @franzliedke has stated will happen eventually).
2. Move to the customised FluxBB fork by @miovisman which already runs on newer PHP versions. This requires more work though as there are features in that bundle that we don’t use and would affect our current FluxBB code customisations.

**Comment by MioVisman** [comment]
FluxBB 1.5.11+ https://github.com/fluxbb/fluxbb/archive/master.zip
and fix for PHP 7.4 https://fluxbb.org/development/core/tickets/1138/ (https://github.com/fluxbb/fluxbb/pull/233)
and  fix for PHP 8.0 https://fluxbb.org/development/core/tickets/1151/
and  PHP 8.0
= work OK

**Comment by philwareham** [comment]
@MioVisman that’s great stuff, thank you very much. Super helpful.

I’ll roll this into our FluxBB fork next week and test locally in MAMP running PHP 8.0.1. Will report back.

**Comment by MioVisman** [comment]
+ https://fluxbb.org/development/core/tickets/1147/ need

**Comment by philwareham** [comment]
@petecooper @Bloke I've now incorporated all the above fixes into our custom FluxBB codebase and tested locally in PHP 8.0.1. I've got an error in the SpamAssassin code that I need to debug (probably my fault with some copy/paste) but the above issue is pretty much closed now.

TLDR; we can soon migrate the forum to the new server.

**Comment by philwareham** [comment]
Many thanks again @MioVisman

**Comment by franzliedke** [comment]
@philwareham Sorry for not responding earlier. 

I now own the fluxbb.org domain and am currently moving the site to a new server. Please do get in touch (preferably via email) to let me know about your plans and how I can best assist you.

Best, Franz

**Comment by philwareham** [comment]
@franzliedke thanks, I've got FluxBB running locally in PHP 8.0.1 now thanks to suggestions from MioVisman. I just need to fix the SpamBarrier code I'd added to @register.php@ as I seem to have broken that with the code updates I've introduced (my fault I think, not a problem with the software).

Good luck with the server move, and hope FluxBB continues to thrive!

**Comment by MioVisman** [comment]
Found an error that occurs when searching for a guest if nothing is found. Available in the original FluxBB and in my version of the engine.
Description and solution: https://fluxbb.org/development/core/tickets/1151/#uqks85z54944

**Comment by philwareham** [comment]
Thanks for the info @MioVisman - I'll amend our forum code as suggested.

**Comment by petecooper** [comment]
@philwareham - please advise if anything is changed in murof and I'll merge into final destination prior to migration.

**Comment by philwareham** [comment]
Done and tested (works as advertised). Also copied updated file to our 'murof' instance in prep for final go-live.

**Comment by petecooper** [comment]
All migrated. Thanks again @MioVisman - it is working perfectly.

**Comment by Neustradamus** [comment]
I have recalled the fluxbb team to add new team members to solve all in upstream.
